### PR TITLE
[ARCTIC-629][SPARK]Fix spark update has no filter will throw exception

### LINKED ISSUE #629: [Bug]: Spark sql update/delete if has no filter will throw exception
### What happened?

If the filter is not added to the update/delete sql, it will throw exception

### Affects Versions

0.3.2

### What engines are you seeing the problem on?

Spark

### How to reproduce

these sqls will throw exception:
update table set col = 'xxx'
delete from table

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hellojinsilei**
## Why are the changes needed?

fix #629 
when update or delete has no filter, will throw exception

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]
LGTM

**Comment by zhoujinsong** [comment]
@hellojinsilei Is there a issue about this bug? If not you better be able create one so user can find that when meeting this bug.

File: spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/optimize/RewriteDeleteFromArcticTable.scala [hunk 1]
```diff
@@ -40,7 +40,12 @@ case class RewriteDeleteFromArcticTable(spark: SparkSession) extends Rule[Logica
       val r = asTableRelation(table)
       val upsertWrite = r.table.asUpsertWrite
       val scanBuilder = upsertWrite.newUpsertScanBuilder(r.options)
-      pushFilter(scanBuilder, condition.get, r.output)
+      if (condition.isEmpty) {
+        val cond = EqualTo(Literal(1), Literal(1))
```
Comment (pan3793) [inline spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/optimize/RewriteDeleteFromArcticTable.scala:44]: `Literal.TrueLiteral` is more straightforward

File: spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/optimize/RewriteDeleteFromArcticTable.scala [hunk 2]
```diff
@@ -40,7 +40,12 @@ case class RewriteDeleteFromArcticTable(spark: SparkSession) extends Rule[Logica
...
+        pushFilter(scanBuilder, cond, r.output)
+      } else {
+        pushFilter(scanBuilder, condition.get, r.output)
+      }
```
Comment (pan3793) [inline spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/optimize/RewriteDeleteFromArcticTable.scala:48]:
```suggestion
      condition match {
        case Some(cond) => pushFilter(scanBuilder, condition.get, r.output)
        case None => pushFilter(scanBuilder, Literal.TrueLiteral, r.output)
      }
```
